### PR TITLE
[WIP] LTP fallback installation

### PR DESCRIPTION
**Pull request by pevik**
TL;DR: poo#41066
+ there are other 3 minor commits

When running LTP on VM (qemu, svirt backend) ltp_* jobs which runs
actual testing are run after install_ltp, they have set:

START_AFTER_TEST=install_ltp+%DISTRI%+%FLAVOR%,install_ltp+%DISTRI%+%VERSION%+%FLAVOR%

But this is failing on baremetal/IPMI due FOLLOW_TEST_DIRECTLY not yet
implemented (see poo#41066), therefore use this workaround.

This required to put most of the install_ltp.pm code outside into
lib/LTP/Install.pm. And also handle tests requiring extra kernel
parameters added via GRUB_PARAM variable in add_custom_grub_entries().

Verification run:
Unfortunately not tested on IPMI (missing worker)
* SLE12 intel
http://quasar.suse.cz/tests/3114
http://quasar.suse.cz/tests/3117
http://quasar.suse.cz/tests/3119
http://quasar.suse.cz/tests/3129

* Tumbleweed intel
http://quasar.suse.cz/tests/3113
http://quasar.suse.cz/tests/3121
http://quasar.suse.cz/tests/3120

* SLE12 s390x
http://quasar.suse.cz/tests/3122
http://quasar.suse.cz/tests/3123

### CODE REVIEW

**Comment by pevik** [comment]
@richiejp @mimi1vx @baierjan can you please review it? I'll add verification run for qemu ASAP.

File: tests/kernel/boot_ltp.pm [hunk 1]
```diff
@@ -45,6 +47,16 @@ sub run {
 
     assert_script_run('export LTPROOT=/opt/ltp; export LTP_COLORIZE_OUTPUT=n TMPDIR=/tmp PATH=$LTPROOT/testcases/bin:$PATH');
 
+    if (script_run('test -d $LTPROOT') != 0) {
```
Comment (richiejp) [inline tests/kernel/boot_ltp.pm]: I think this should only be enabled for scenarios where you expect it to happen. We don't want someone to accidentally schedule 100 LTP installations by configuring an LTP test with the wrong image.

Comment (pevik) [reply tests/kernel/boot_ltp.pm]: OK, I'll put it only for baremetal tests (`LTP_BAREMETAL == 1`).

Comment (pevik) [reply tests/kernel/boot_ltp.pm]: @richiejp on the other hand we agreed with metan to switch to LTP installing from package (there are nightly builds, so it saves time just to install it).

Comment (richiejp) [reply tests/kernel/boot_ltp.pm]: Yeah, the VM image grows quite a lot during LTP compilation/installation, so it might be best to remove that and install from the package on every test. That could fix the boot issue on Tumbleweed x86.

File: lib/LTP/Install.pm [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+    assert_script_run('which ping6 >/dev/null 2>&1 || ln -s `which ping` /usr/local/bin/ping6');
+
+    # dhcpd
+    assert_script_run('touch /var/lib/dhcp/db/dhcpd.leases /var/lib/dhcp6/db/dhcpd6.leases');
+
+    # echo/echoes, getaddrinfo_01
+    assert_script_run('sed -i \'s/^\(hosts:\s+files\s\+dns$\)/\1 myhostname/\' /etc/nsswitch.conf');
+
+    # SLE12GA uses too many old style services
+    my $action = check_var('VERSION', '12') ? "enable" : "reenable";
```
Comment (baierjan) [inline lib/LTP/Install.pm]: As the 12GA is not supported anymore, this line could be dropped and the code cleaned a little bit?

Comment (pevik) [reply lib/LTP/Install.pm]: @baierjan good point, sent as separate PR: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/7818

File: tests/kernel/boot_ltp.pm (hunk 1 above)
Comment (pevik) [reply tests/kernel/boot_ltp.pm]: @richiejp ok, I'll do 2 things: 1) install LTP from package (https://progress.opensuse.org/issues/53951, but keep possibility to install from git) and then implement this (install LTP on each test).

**Comment by pevik** [comment]
Putting as WIP as after switching to installing LTP from package (poo#53951) it might be better to run install_ltp before each test as Richie suggested.

**Comment by pevik** [comment]
Closing this. After implementing https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/7865 we agreed that better will be running `install_ltp.pm` in each test. While it's now only 2x faster than compilation (so running in each test would take a lot), some time will be saved also by avoiding publishing image with LTP installation and installing runtime dependencies via package dependencies (https://progress.opensuse.org/issues/54506).

**Comment by pevik** [comment]
Martin did much better solution in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8329, depends on https://github.com/os-autoinst/openQA/pull/2302.
